Source/Generic/ECTestCase.h

```diff
@@ -9,6 +9,18 @@
 
 @class NSBitmapImageRep;
 
+/// The reporting mode enum when comparing test results
+/// Are differences reported by character, line etc?
+typedef NS_ENUM(NSUInteger, ECTestComparisonMode)
+{
+	ECTestComparisonShowChars,
+	ECTestComparisonShowLines,
+	ECTestComparisonShowLinesIgnoreWhitespace,
+	ECTestComparisonDiff,
+	ECTestComparisonDiffNoJSON,
+	ECTestComparisonDefault
+};
```

Comment:
This all looks very interesting. I'd ideally like to have some docs for each constant, but maybe that's not for this pull.